views/apps/new.html

```diff
@@ -1,241 +1,279 @@
-      dateInput.value = null;
-      expiryOptions.classList.add("hidden");
-    } else {
-      dateInput.value = sixMonthsFromNow.toISOString().split('T')[0];
-      expiryOptions.classList.remove("hidden");
-    }
-  });
+  var requestMethodsPopUp = document.getElementById("RequestMethodsPopUp");
+  const requestMethodCheckboxes = document.getElementById("RequestMethodOptions").querySelectorAll("input[type=checkbox]");
+  requestMethodCheckboxes.forEach(checkbox => {
+    checkbox.addEventListener("change", function(e) {
+      let checkedValues = "";
+      requestMethodCheckboxes.forEach(checkbox => {
+        if (checkbox.checked) {
+          checkedValues += checkbox.value + " ";
+        }
+      });
+      requestMethodsPopUp.value = checkedValues.trim();
+    })
+  })
 
   var budgetCheckbox = document.getElementById("BudgetCheckbox");
   var budgetOptions = document.getElementById("BudgetOptions");
-  var budgetInput = document.getElementById("MaxAmount");
+  var budgetInput = document.getElementById("MaxAmountPopUp");
+  const budgetOptionRadios = document.getElementById("BudgetOptions").querySelectorAll("input[type=radio]");
 
   budgetCheckbox.addEventListener("change", function(e) {
     if (!budgetCheckbox.checked) {
       budgetInput.value = null;
       budgetInput.required = false;
-      budgetOptions.classList.add("hidden");
+      budgetInput.disabled = true;
+      budgetOptions.classList.add("opacity-30");
+      budgetOptionRadios.forEach(radio => {
+        radio.disabled = true;
+        radio.checked = radio.value === "never";
+      })
     } else {
       budgetInput.required = true;
-      budgetOptions.classList.remove("hidden");
+      budgetInput.disabled = false;
+      budgetOptionRadios.forEach(radio => {
+        radio.disabled = false;
+      })
+      budgetOptions.classList.remove("opacity-30");
     }
   });
 
-  const requestMethodCheckboxes = document.getElementById("RequestMethodOptions").querySelectorAll("input[type=checkbox]");
-  requestMethodCheckboxes.forEach(checkbox => {
-    checkbox.addEventListener("change", function(e) {
-      let checkedValues = "";
-      requestMethodCheckboxes.forEach(checkbox => {
-        if (checkbox.checked) {
-          checkedValues += checkbox.value + " ";
-        }
-      });
-      document.getElementById("RequestMethods").value = checkedValues.trim();
+  var saveSettings = document.getElementById("save-settings");
+  saveSettings.addEventListener("click", function(e) {
+    var requestMethods = document.getElementById("RequestMethods");
+    requestMethods.value = requestMethodsPopUp.value;
+
+    dateInput.value = dateInputPopUp.value;
+
+    var budgetRenewal = document.getElementById("BudgetRenewal");
+    budgetOptionRadios.forEach(radio => {
+      if (radio.checked) {
+        budgetRenewal.value = radio.value
+      }
     })
+
+    var maxAmount = document.getElementById("MaxAmount");
+    maxAmount.value = budgetInput.value;
```

Comment:
can you make the naming match again here so it is consistent?
either `maxAmount.value = maxAmountPopup.value`

And maybe `FromPopup` would make more sense than just `Popup` for all?

Reply:
Done